src/core/query.ts

```diff
@@ -365,11 +368,8 @@ export class Query<
       }
     }
 
-    if (
-      process.env.NODE_ENV !== 'production' &&
-      !Array.isArray(this.options.queryKey)
-    ) {
-      getLogger().error(
+    if (!Array.isArray(this.options.queryKey)) {
+      this.logger.error(
```

Comment:
I understand why you made that change, since the default logger is now stubbed for production. However, if you pass in your own logger, you will now get these logs also in production unless you make the env check yourselves.

Reply:
Which is the point of custom loggers, requiring finer control on logging and logging levels in different environments. For example, @userlike, we set log levels to "error" and catch console errors with sentry that happens in production.